bundle/edu.gemini.spModel.core/src/main/scala/edu/gemini/spModel/core/Epoch.scala

```diff
@@ -1,18 +1,152 @@
 package edu.gemini.spModel.core
 
-import scalaz._, Scalaz._
+import java.time.ZoneOffset.UTC
+import java.time._
 
-/** Epoch in Gregorian (?) years. */
-final case class Epoch(year: Double)
+import scalaz._
+import Scalaz._
 
+/**
+ * An epoch, the astronomer's equivalent of `Instant`, based on a fractional year in some temporal
+ * scheme (Julian, in this case) that determines year zero and the length of a year. The only
+ * meaningful operation for an `Epoch` is to ask the elapsed epoch-years between it and some other
+ * point in time. We need this for proper motion corrections because velocities are measured in
+ * motion per epoch-year. The epoch year is stored internally as integral milliyears.
```

Comment:
Why is `integral milliyears` but the type in `Epoch` is a `Double`?

Reply:
In OCS3, we're storing as integral milliyears, but in OCS2, we store as `Double`? I'll fix this comment.